sdk/azcore/cloud/cloud.go

```diff
@@ -0,0 +1,192 @@
+//go:build go1.16
+// +build go1.16
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package cloud
+
+import "encoding/json"
+
+// WellKnownClouds contains configuration settings for public Azure clouds.
+var WellKnownClouds = map[Name]Configuration{}
+
+// Name identifies a cloud.
+type Name string
+
+const (
+	// AzureChina is a global constant identifying Azure China.
+	AzureChina Name = "AzureChina"
+	// AzureGovernment is a global constant identifying Azure Government.
+	AzureGovernment Name = "AzureGovernment"
```

Comment:
I think this should be `AzureUSGovernment` consistent with the naming in other tooling, and also `AzureGovernment` isn't really specific if you're not US-based anyway.

Reply:
`AzureUSGovernment` is a good name and consistent with ARM metadata at least. I'm not sure it's the right name though. Our GA Identity libraries have an `AzureGovernment` constant because the cloud is called "Azure Government" in docs. I also doubt we want to prioritize consistency with ARM metadata. That would mean having `AzureCloud` instead of `AzurePublicCloud`. The latter isn't a great name but it's clearer than the former. I chose to start with inter-SDK consistency and intend to raise this with architects.